From 3488abc2c32dd00c21d37980886a8d7bee100ea7 Mon Sep 17 00:00:00 2001
From: xiaoton1 <xiaotong.wang@intel.com>
Date: Fri, 5 Aug 2022 12:40:32 +0800
Subject: [PATCH] FROMGIT: iHD: [VP] Force set m_EngineCaps to 0 in construct
 function if not initialization list not working well

Ensure m_EngineCaps 0 if initialization list not working in compiler.

(cherry picked from commit 84601ad18ce7fd7f484752c2db55cb1b49079cb0 https://github.com/intel/media-driver.git)

BUG=237612196
TEST= 1. Hold Google Meet 1:1 call 1.1 Make sure the chromebook send VP9 stream.
      2. On the remote (recieving) end, resize the window to alternately force QVGA or GVA resolution.

Signed-off-by: Gareth Yu <gareth.yu@intel.com>
---
 .../agnostic/common/vp/hal/feature_manager/sw_filter.cpp    | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/media_softlet/agnostic/common/vp/hal/feature_manager/sw_filter.cpp b/media_softlet/agnostic/common/vp/hal/feature_manager/sw_filter.cpp
index 0c687aa..df0b053 100644
--- a/media_softlet/agnostic/common/vp/hal/feature_manager/sw_filter.cpp
+++ b/media_softlet/agnostic/common/vp/hal/feature_manager/sw_filter.cpp
@@ -46,6 +46,12 @@ inline void swap(T &a, T &b)
 
 SwFilter::SwFilter(VpInterface &vpInterface, FeatureType type) : m_vpInterface(vpInterface), m_type(type)
 {
+    if (m_EngineCaps.value != 0)
+    {
+        // Some complier may not work well for m_EngineCaps initialization by m_EngineCaps = {}. Force set to 0 here.
+        VP_PUBLIC_NORMALMESSAGE("m_EngineCaps is not initialized correctly since complier issue, m_EngineCaps.value: %x. Force reset to 0.", m_EngineCaps.value);
+        m_EngineCaps.value = 0;
+    }
 }
 
 SwFilter::~SwFilter()
-- 
2.31.0

